lib/graphql_metrics/extractor.rb

```diff
@@ -2,117 +2,58 @@
 
 module GraphQLMetrics
   class Extractor
-    CONTEXT_NAMESPACE = :extracted_metrics
-    TIMING_CACHE_KEY = :timing_cache
-    START_TIME_KEY = :query_start_time
-
     EXPLICIT_NULL = 'EXPLICIT_NULL'
     IMPLICIT_NULL = 'IMPLICIT_NULL'
     NON_NULL = 'NON_NULL'
 
-    attr_reader :query, :ctx_namespace
-
-    def self.use(schema_definition)
-      extractor = self.new
-      return unless extractor.extractor_defines_any_visitors?
-
-      extractor.setup_instrumentation(schema_definition)
-    end
-
-    def use(schema_definition)
-      return unless extractor_defines_any_visitors?
-      setup_instrumentation(schema_definition)
-    end
+    attr_reader :query, :instrumentor
 
-    def setup_instrumentation(schema_definition)
-      schema_definition.instrument(:query, self)
-      schema_definition.instrument(:field, self)
-    end
-
-    def before_query(query)
-      return unless extractor_defines_any_visitors?
-
-      ns = query.context.namespace(CONTEXT_NAMESPACE)
-      ns[TIMING_CACHE_KEY] = {}
-      ns[START_TIME_KEY] = Process.clock_gettime(Process::CLOCK_MONOTONIC)
-    rescue StandardError => ex
-      handle_extraction_exception(ex)
+    def initialize(instrumentor)
```

Comment:
Doesn't this need 

```suggestion
    def initialize(instrumentor = nil)
```

It can't be instantiated without one right now. I guess it's meant to be overridden, but still seems like this is a better default to have? Not every extractor might have a need on its own to override `initialize`.